src/ImageSharp.Web/FormatUtilities.cs

```diff
@@ -51,14 +51,26 @@ public FormatUtilities(IOptions<ImageSharpMiddlewareOptions> options)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public string GetExtensionFromUri(string uri)
         {
+            // TODO: This method should follow TryGet pattern. Fix for V2.
             int query = uri.IndexOf('?');
             ReadOnlySpan<char> path;
 
             if (query > -1)
             {
-                if (uri.Contains(FormatWebProcessor.Format, StringComparison.OrdinalIgnoreCase) && QueryHelpers.ParseQuery(uri.Substring(query)).TryGetValue(FormatWebProcessor.Format, out StringValues ext))
+                if (uri.Contains(FormatWebProcessor.Format, StringComparison.OrdinalIgnoreCase)
+                    && QueryHelpers.ParseQuery(uri.Substring(query)).TryGetValue(FormatWebProcessor.Format, out StringValues ext))
                 {
-                    return ext;
+                    // We have a query but is it a valid one?
+                    ReadOnlySpan<char> extSpan = ext[0].AsSpan();
+                    foreach (string extension in this.extensions)
```

Comment:
You might get better performance here if the extensions list was a hashset (might :))

extensions would always be unique right?

Reply:
Yeah would always be unique

Reply:
Yeah I agree, it would be very marginal- just saw it and thought hashset. All good